cmd/limactl/prune.go

```diff
@@ -1,30 +1,172 @@
 package main
 
 import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
+	"github.com/lima-vm/lima/pkg/store"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
 func newPruneCommand() *cobra.Command {
 	pruneCommand := &cobra.Command{
-		Use:               "prune",
-		Short:             "Prune garbage objects",
-		Args:              WrapArgsError(cobra.NoArgs),
-		RunE:              pruneAction,
-		ValidArgsFunction: cobra.NoFileCompletions,
+		Use:   "prune",
+		Short: "Prune garbage objects",
+		RunE:  pruneAllAction, // keep old behavior
 	}
+
+	pruneCommand.AddCommand(
+		newPruneAllCommand(),
+		newPruneDownloadsCommand(),
+	)
+
 	return pruneCommand
 }
 
-func pruneAction(cmd *cobra.Command, args []string) error {
-	ucd, err := os.UserCacheDir()
+func newPruneAllCommand() *cobra.Command {
+	pruneAllCommand := &cobra.Command{
+		Use:   "all",
+		Short: "Prune all caches",
+		RunE:  pruneAllAction,
+	}
+	return pruneAllCommand
+}
+
+func pruneAllAction(cmd *cobra.Command, args []string) error {
+	cacheDir, err := getCacheDir()
 	if err != nil {
 		return err
 	}
-	cacheDir := filepath.Join(ucd, "lima")
-	logrus.Infof("Pruning %q", cacheDir)
+	logrus.Infof("Pruning everything in %q", cacheDir)
 	return os.RemoveAll(cacheDir)
 }
+
+func newPruneDownloadsCommand() *cobra.Command {
+	pruneDownloadsCommand := &cobra.Command{
+		Use:   "downloads",
+		Short: "Prune unreferenced downloads",
+		RunE:  pruneDownloadsAction,
+	}
```

Comment:
How is "download" different from "cache"?

Reply:
We don't have it YET :) but why do we have "download" directory under the cache then? ;)